builder/azure/arm/artifact.go

```diff
@@ -133,7 +133,10 @@ func (*Artifact) Files() []string {
 }
 
 func (a *Artifact) Id() string {
-	return a.OSDiskUri
+	if a.OSDiskUri != "" {
```

Comment:
Please add a test case for this.  I am sorry I missed this the first time around, but no reason to repeat the mistake again if we can avoid it.